config.py

```diff
@@ -1,10 +1,86 @@
-Size = 20  # C 20 The size of the sides of each square (in pixels)
-Cushion = 5  # C 10 How far the board extends beyond the visible amount
-Edge = maths.ceil(Size / 7)  # C Size/15 The gap between each cell
-Background = (120, 120, 120)  # C (120,120,120) The colour of the background
+import pygame
+
 Dead = 0
 Square = 1
+
+class cell:
+    def __init__(self):
+        self.CurrentState = Square
+        self.NextState = Dead
+
+    def kill(self):
+        self.NextState = Dead
+
+    def birth(self, state):
+        self.NextState = state
+
+
+class Board:
+    def __init__(self):  # customisable, default_value, description
+        self.Width = 50  # C 50 How many squares wide the board is
+        self.Height = 30  # C 30 Ditto but with height
+        self.Size = 20  # C 20 The size of the sides of each square (in pixels)
+        self.Wrap = False  # Whether the board wraps around on itself
+        self.Edge = self.Size // 7  # C self.Size / 15 The gap between each cell
+        self.Generations = 0
+        if self.Wrap:
+            self.Cushion = 0  # C 10 How far the board extends beyond the visible amount
+        else:
+            self.Cushion = 10
+        self.Cell = [[cell() for _ in range(self.Height + (2 * self.Cushion))] for _ in range(self.Width + 2 *
+                                                                                              self.Cushion)]
+B = Board()
+
+
+class GameState:
+    def __init__(self):
+        self.GPS = 10  # C 10 How many Generations Per Seconds
+        self.TopGPS = 100  # The GPS at the top of the slider.
+        self.BottomGPS = 0.5  # The GPS at the bottom of slider.
+        self.GPSLimit = True
+        self.Paused = True
+        self.OneTurn = False
+        self.Colour = {"Alive": (0, 0, 0),
+                       "Player1": (0, 255, 100),
+                       "Player2": (0, 100, 255),
+                       "Player3": (100, 255, 0),
+                       "Player4": (100, 0, 255),
+                       "Dead": (255, 255, 255),
+                       "Highlighter": (0, 255, 100),
+                       "Background": (120, 120, 120),
+                       "Text": (180, 180, 180),
+                       "Unselected": (160, 160, 160)}
+
+
+class Widgets:
+    def __init__(self):
+        self.NoOfButtons = 0
+        self.ButtonSize = 50
+        self.HighlightSize = 5
+        self.NoOfNotches = 9
+        self.NotchLength = self.ButtonSize/5
+        self.StartOfSlider = self.NoOfButtons * self.ButtonSize + self.HighlightSize + 2 * self.NotchLength
+        self.EndOfSlider = B.Height * B.Size - self.HighlightSize - self.NotchLength
+        self.SpaceBetweenNotches = (self.EndOfSlider-self.StartOfSlider) / (self.NoOfNotches-1)
+        self.ButtonStart = B.Size * B.Width
+        self.SliderY = B.Size * B.Width + B.Edge / 2 + self.ButtonSize / 2
+
+
+def write(screen, x, y, text, colour, size, rotate=0, alignment=("left", "top")):
```

Comment:
Again, this file is for constant variables, so that the user could go into this file and change stuff. The user probably won't want to change the function itself. You can make an `alignment` variable here, and then use it in the main game file.

Reply:
done